libsolidity/interface/CompilerStack.cpp

```diff
@@ -194,7 +217,30 @@ bool CompilerStack::parse()
 				else
 					noErrors = false;
 
-				m_contracts[contract->name()].contract = contract;
+				if (m_contracts.find(contract->fullyQualifiedName()) != m_contracts.end())
+				{
+					const ContractDefinition* existingContract = m_contracts.find(contract->fullyQualifiedName())->second.contract;
```

Comment:
Perhaps use `m_contracts[contract->fullyQualifiedName()]` here? Also please put the `const` after the type name: `ContractDefinition const*`